bob-apps/bin/blob_recovery_tool/command.rs

```diff
@@ -0,0 +1,285 @@
+use super::prelude::*;
+
+const INPUT_OPT: &str = "input blob";
+const OUTPUT_OPT: &str = "output blob";
+const VALIDATE_EVERY_OPT: &str = "record cache";
+const DISK_PATH_OPT: &str = "disk path";
+const SUFFIX_OPT: &str = "suffix";
+const BACKUP_SUFFIX_OPT: &str = "backup suffix";
+const FIX_OPT: &str = "fix";
+const NO_CONFIRM_OPT: &str = "no confirm";
+const DELETE_OPT: &str = "index delete";
+
+const VALIDATE_INDEX_COMMAND: &str = "validate-index";
+const VALIDATE_BLOB_COMMAND: &str = "validate-blob";
+const RECOVERY_COMMAND: &str = "recovery";
+
+pub enum MainCommand {
+    Recovery(RecoveryBlobCommand),
+    Validate(ValidateBlobCommand),
+    ValidateIndex(ValidateIndexCommand),
+}
+
+pub struct RecoveryBlobCommand {
+    input: String,
+    output: String,
+    validate_every: usize,
+}
+
+impl RecoveryBlobCommand {
+    fn run(&self) -> AnyResult<()> {
+        recovery_blob(&self.input, &self.output, self.validate_every)
+    }
+
+    fn subcommand<'a, 'b>() -> App<'a, 'b> {
+        SubCommand::with_name(RECOVERY_COMMAND)
+            .arg(
+                Arg::with_name(INPUT_OPT)
+                    .help("input blob")
+                    .takes_value(true)
+                    .required(true)
+                    .short("i")
+                    .long("input"),
+            )
+            .arg(
+                Arg::with_name(OUTPUT_OPT)
+                    .help("output blob")
+                    .takes_value(true)
+                    .required(true)
+                    .short("o")
+                    .long("output"),
+            )
+            .arg(
+                Arg::with_name(VALIDATE_EVERY_OPT)
+                    .help("validate every N records")
+                    .takes_value(true)
+                    .default_value("100")
+                    .short("c")
+                    .value_name("N")
+                    .long("cache-size"),
+            )
+    }
+
+    fn from_matches(matches: &ArgMatches) -> AnyResult<RecoveryBlobCommand> {
+        Ok(RecoveryBlobCommand {
+            input: matches.value_of(INPUT_OPT).expect("Required").to_string(),
+            output: matches.value_of(OUTPUT_OPT).expect("Required").to_string(),
+            validate_every: matches
+                .value_of(VALIDATE_EVERY_OPT)
+                .expect("Has default")
+                .parse()?,
+        })
+    }
+}
+
+pub struct ValidateBlobCommand {
+    path: PathBuf,
+    blob_suffix: String,
+    backup_suffix: String,
+    validate_every: usize,
+    fix: bool,
+    confirm: bool,
```

Comment:
It's not clear what's the purpose of this field. And why in args it's no-confirm